scripts/datastore_upgrade.py

```diff
@@ -0,0 +1,311 @@
+                cell_values_to_update[entity_key] = cell_value
+    try:
+        datastore.batch_put_entity(APP_ENTITY_TABLE, row_keys_to_update, APP_ENTITY_SCHEMA, cell_values_to_update)
+        print("Batch put entities was successful")
+    except  AppScaleDBConnectionError as error:
+        print("Exception on batch put entity: {}".format(str(error)))
+        exit(1)
+
+def is_txn_id_same(entity, key, valid_entity):
+    """ Compares if the transaction id is the same for the current and
+    validated entity.
+    Args:
+        entity: Current tombstonedless entity.
+        key: A key to look for in the validated entity.
+        valid_entity: A validated entity to compare with.
+    Returns:
+        True if transaction id is the same.
+    """
+    return entity[key][APP_ENTITY_SCHEMA[1]] == valid_entity[key][APP_ENTITY_SCHEMA[1]]
+
+def is_entity_same(entity, key, valid_entity):
+    """ Compares if the transaction id is the same for the current and
+        validated entity.
+    Args:
+        entity: Current tombstonedless entity.
+        key: A key to look for in the validated entity.
+        valid_entity: A validated entity to compare with.
+    Returns:
+        True is the encoded entity is the same.
+    """
+    return entity[key][APP_ENTITY_SCHEMA[0]] == valid_entity[key][APP_ENTITY_SCHEMA[0]]
+
+def compare_and_get_invalid_keys(current_entities, valid_entity_keys):
+    """ Looks for the current entities keys in a list of valid entity keys
+    and returns the invalid keys.
+    Args:
+        current_entities: A list of entities currently in APP_ENTITY_TABLE.
+        key: A key to look for in the validated entity.
+        valid_entity_keys: A list valid entity keys to compare from.
+    Returns:
+        A list of invalid entity keys
+    """
+    invalid_entity_keys = []
+    for entity in current_entities:
+        key = entity.keys()[0]
+        if key not in valid_entity_keys:
+            if key not in invalid_entity_keys:
+                invalid_entity_keys.append(key)
+    return invalid_entity_keys
+
+def get_valid_entity_keys(validated_entities):
+    """ Loops through the validated entities and creates a list of valid
+    entity keys.
+    Args:
+        validated_entities: A list of validated entities.
+    Returns:
+        A list of valid entity keys.
+    """
+    valid_row_keys = []
+    # Get all the keys from the validated entities
+    for valid_entity in validated_entities:
+        valid_key = valid_entity.keys()[0]
+        if valid_key not in valid_row_keys:
+            valid_row_keys.append(valid_key)
+    return valid_row_keys
+
+def get_zk_locations_string(zk_location_ips):
+    """ Generates a ZooKeeper IP locations string.
+    Args:
+        zk_location_ips: A list of ZooKeeper IP locations.
+    Returns:
+        A string of ZooKeeper IP locations
+    """
+    return (ZK_PORT + ",").join(zk_location_ips) + ZK_PORT
```

Comment:
You'll probably want to use an integer for ZK_PORT so that you can format it with a colon when necessary and use it in cases when it doesn't need a colon.